grails-rundeck-data-shared/src/main/groovy/rundeck/data/util/WorkflowStepUtil.groovy

```diff
@@ -0,0 +1,37 @@
+package rundeck.data.util
+
+import org.rundeck.app.data.model.v1.job.workflow.WorkflowStepData
+import org.rundeck.core.execution.BaseCommandExec
+import rundeck.data.constants.WorkflowStepConstants
+
+class WorkflowStepUtil {
+
+    static String summarize(WorkflowStepData step) {
+        if(step.pluginType == WorkflowStepConstants.TYPE_COMMAND & step instanceof BaseCommandExec) {
+            return summarizeCommandStep(step)
+        } else if(step.pluginType == WorkflowStepConstants.TYPE_JOB_REF) {
+            return summarizeJobStep(step)
+        } else {
+            return "Plugin["+ step.pluginType + ', nodeStep: '+step.nodeStep+']'
+        }
+
+    }
+
+    static String summarizeJobStep(WorkflowStepData step) {
+        return "job: ${step.getJobIdentifier()}${step.argString?' -- '+step.argString:''}"
+    }
+
+    static String summarizeCommandStep(BaseCommandExec step){
```

Comment:
Since the RdWorkflowStep implementation doesn't implement BaseCommandExec you'll either need to convert the RdWorkflowStep to a BaseCommandExec or not use that type here.